clients/web/src/views/widgets/DateTimeWidget.js

```diff
@@ -0,0 +1,228 @@
+/* globals moment */
+
+/**
+ * This widget provides a text input field to specify a date/time. The user
+ * chooses the date/time using a popup picker.
+ */
+girder.views.DateTimeWidget = girder.View.extend({
+
+    /**
+     * @param [settings.prefix='default'] Prefix for element IDs in case
+     *     multiple DateTimeWidgets are rendered simultaneously.
+     */
+    initialize: function (settings) {
+        this.prefix = settings.prefix || 'default';
+        this.dateTimeId = '#' + this.prefix + '-datetime';
```

Comment:
We generally try to avoid using id's unless we have to. In this case I think we could get away with a fixed class instead, like `.g-datetime`, and use `this.$('.g-datetime')` to select it without having to worry about the global uniqueness problem.

Also, for both of these widgets, let's make the initial date value something that can be passed in the settings object to `initialize`.